File: 1646/Solution.ts
const getMaximumGenerated = (n: number): number => {
    const nums: number[] = []
    const max = (n - 1) / 2
    nums[0] = 0
    nums[1] = 1
    for (let i = 1; i <= max; i++) {
        nums[2 * i] = nums[i]
        nums[2 * i + 1] = nums[i] + nums[i + 1]
    }
    return Math.max(...nums)
}

console.log(getMaximumGenerated(7))

File: 1646/Solution.cpp
#include "../LeetCode.h"
using namespace lc;

auto Solution::getMaximumGenerated(int n) -> int
{
    std::vector<int> nums;
    int max = (n - 1) / 2;
    nums.push_back(0);
    for (int i = 1; i <= n; i++) nums.push_back(1);
    for (int i = 1; i <= max; i++)
    {
        nums[2 * i] = nums[i];
        nums[2 * i + 1] = nums[i] + nums[i + 1];
    }
    return *max_element(nums.begin(),nums.end());
}